### ISSUE TITLE
Link color is hard to read in dark mode with "black" primary color 

### ISSUE THREAD

**Issue by willmcgugan** [squidfunk/mkdocs-material#4600]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

The link color with black primary color, is a little hard to read in a recent update.

<img width="649" alt="Screenshot 2022-11-06 at 14 15 55" src="https://user-images.githubusercontent.com/554369/200177740-5e9cba4d-b9f2-4f55-8723-2ff3aafe732c.png">

### Expected behaviour

Link color should be a little lighter.

### Actual behaviour

Link color doesn't stand out from background.

### Steps to reproduce

Set primary color to "black"

### Package versions

- Python: `python --version`
- MkDocs: `mkdocs --version`
- Material: `pip show mkdocs-material | grep -E ^Version`
3.10
14.2
Version: 8.5.7+insiders.4.26.1

### Configuration

```yaml
site_name: Textual
site_url: https://textual.textualize.io/
repo_url: https://github.com/textualize/textual/
edit_uri: edit/main/docs/

nav:
  - Introduction:
    - "index.md"
    - "getting_started.md"
    - "help.md"
  - "tutorial.md"
  - Guide:
    - "guide/index.md"
    - "guide/devtools.md"
    - "guide/app.md"
    - "guide/styles.md"
    - "guide/CSS.md"
    - "guide/queries.md"
    - "guide/layout.md"
    - "guide/events.md"
    - "guide/input.md"
    - "guide/actions.md"
    - "guide/reactivity.md"
    - "guide/widgets.md"
    - "guide/animation.md"
    - "guide/screens.md"
  - "roadmap.md"
  - Reference:
    - Events:
      - "events/index.md"
      - "events/blur.md"
      - "events/descendant_blur.md"
      - "events/descendant_focus.md"
      - "events/enter.md"
      - "events/focus.md"
      - "events/hide.md"
      - "events/key.md"
      - "events/leave.md"
      - "events/load.md"
      - "events/mount.md"
      - "events/mouse_capture.md"
      - "events/click.md"
      - "events/mouse_down.md"
      - "events/mouse_move.md"
      - "events/mouse_release.md"
      - "events/mouse_scroll_down.md"
      - "events/mouse_scroll_up.md"
      - "events/mouse_up.md"
      - "events/paste.md"
      - "events/resize.md"
      - "events/screen_resume.md"
      - "events/screen_suspend.md"
      - "events/show.md"
    - Styles:
      - "styles/index.md"
      - "styles/align.md"
      - "styles/background.md"
      - "styles/border.md"
      - "styles/box_sizing.md"
      - "styles/color.md"
      - "styles/content_align.md"
      - "styles/display.md"
      - "styles/dock.md"
      - "styles/grid.md"
      - "styles/height.md"
      - "styles/layer.md"
      - "styles/layers.md"
      - "styles/layout.md"
      - "styles/links.md"
      - "styles/margin.md"
      - "styles/max_height.md"
      - "styles/max_width.md"
      - "styles/min_height.md"
      - "styles/min_width.md"
      - "styles/offset.md"
      - "styles/opacity.md"
      - "styles/outline.md"
      - "styles/overflow.md"
      - "styles/padding.md"
      - "styles/scrollbar.md"
      - "styles/scrollbar_gutter.md"
      - "styles/scrollbar_size.md"
      - "styles/text_align.md"
      - "styles/text_style.md"
      - "styles/text_opacity.md"
      - "styles/tint.md"
      - "styles/visibility.md"
      - "styles/width.md"
    - Widgets:
      - "widgets/index.md"
      - "widgets/button.md"
      - "widgets/checkbox.md"
      - "widgets/data_table.md"
      - "widgets/footer.md"
      - "widgets/header.md"
      - "widgets/input.md"
      - "widgets/static.md"
      - "widgets/tree_control.md"
  - API:
    - "api/app.md"
    - "api/button.md"
    - "api/color.md"
    - "api/containers.md"
    - "api/dom_node.md"
    - "api/events.md"
    - "api/footer.md"
    - "api/geometry.md"
    - "api/header.md"
    - "api/index.md"
    - "api/message_pump.md"
    - "api/message.md"
    - "api/pilot.md"
    - "api/query.md"
    - "api/reactive.md"
    - "api/screen.md"
    - "api/static.md"
    - "api/timer.md"
    - "api/widget.md"
  - "Blog":
    - blog/index.md



markdown_extensions:
  - attr_list
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - md_in_html
  - admonition
  - def_list
  - meta

  - toc:
      permalink: true
      baselevel: 1
  - pymdownx.keys
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.highlight:
      anchor_linenums: true
  - pymdownx.inlinehilite
  - pymdownx.superfences:
      custom_fences:
        - name: textual
          class: textual
          format: !!python/name:textual._doc.format_svg
        - name: rich
          class: rich
          format: !!python/name:textual._doc.rich
  - pymdownx.inlinehilite
  - pymdownx.superfences
  - pymdownx.snippets
  - pymdownx.tabbed:
      alternate_style: true
  - pymdownx.snippets
  - markdown.extensions.attr_list

theme:
  name: material
  custom_dir: docs/custom_theme
  features:
    - navigation.tabs
    - navigation.indexes
    - navigation.tabs.sticky
    - content.code.annotate
  palette:
  - media: "(prefers-color-scheme: light)"
    scheme: default
    accent: purple
    toggle:
      icon: material/weather-sunny
      name: Switch to dark mode
  - media: "(prefers-color-scheme: dark)"
    scheme: slate
    primary: black
    toggle:
      icon: material/weather-night
      name: Switch to light mode

plugins:

- blog:
- search:
- autorefs:
- mkdocstrings:
    default_handler: python
    handlers:
        python:
          options:
            show_source: false
            filters:
              - "!^_"
              - "^__init__$"
              - "!^can_replace$"

    watch:
      - src/textual


extra_css:
  - stylesheets/custom.css
```

### System information

- Operating system: ...
- Browser: ...
MacOS Ventura
Chrome

**Comment by squidfunk** [comment]
Thanks for reporting. Regression introduced in https://github.com/squidfunk/mkdocs-material/commit/3c5f94f592aafca8eb6adaed2eb08f6826686b90 to fix #4571.

**Comment by squidfunk** [comment]
Fixed in f5f5baa8d73d356b6789d19feee84080298b56fe.

**Comment by squidfunk** [comment]
Released as part of 8.5.9.
